dipy/io/streamline.py

```diff
@@ -20,7 +27,23 @@ def save_trk(fname, streamlines, affine, vox_size=None, shape=None, header=None)
         The shape of the reference image (default: None)
     header : dict, optional
         Metadata associated to the tractogram file(*.trk). (default: None)
+    reduce_memory_usage : {False, True}, optional
```

Comment:
Even if we go with this name (which I think is fine), I'd still want to make sure that users understand that this will only save memory if they provide an input that is a generator (that's right?), so I would add that to the description.

Reply:
> will only save memory if they provide an input that is a generator (that's right?)

No, whatever the input streamlines object, we do the job under the hood [here](https://github.com/nipy/dipy/pull/1651/files#diff-58c02ec9366478aebd7f3ab2c80a62a6R56) and the user will save memory

Reply:
by saving memory, I mean, it will not increase like a hell 😄

Reply:
But if `streamlines` is a list on input, that wouldn't matter. It would just generate from that list, without saving any memory. Or am I missing something?

Reply:
If you have a list on input,  `Tractogram` create `Streamlines object` which will create a copy of your list and explode the memory for big data. The `reduce_memory_usage` variable allows the conversion of this list in a generator so no copy and no extra memory.

Reply:
Each time you create a `Streamlines`object from a list, a copy is done because they are not organize is the same way on the memory

Reply:
Gotcha - that makes sense. Is there any performance trade-off? Why would anyone ever want to not use this? I mean, if there's no reason to set this to False, should we just eliminate this as an option?

Reply:
Good question! no idea about the performance trade-off.  But in general, the generators are less faster